st2common/tests/unit/test_executions_util.py

```diff
@@ -171,6 +171,19 @@ def test_execution_update(self):
         self.assertGreater(execution.log[1]["timestamp"], pre_update_timestamp)
         self.assertLess(execution.log[1]["timestamp"], post_update_timestamp)
 
+    def test_skip_execution_update(self):
+        liveaction = self.MODELS["liveactions"]["successful_liveaction.yaml"]
+        executions_util.create_execution_object(liveaction)
+        pre_update_status = liveaction.status
+        liveaction.status = "running"
+        executions_util.update_execution(liveaction)
+        execution = self._get_action_execution(
+            liveaction__id=str(liveaction.id), raise_exception=True
+        )
+        self.assertEqual(len(execution.log), 1)
+        # Check status is not updated if it's already in completed state.
+        self.assertEqual(execution.log[0]["status"], pre_update_status)
```

Comment:
Can you add a check to make sure that liveaction/action execution is already completed?